packages/ngtools/webpack/src/transformers/import_factory.ts

```diff
@@ -0,0 +1,198 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+
+
+/**
+ * Given this original source code:
+ *
+ * import { NgModule } from '@angular/core';
+ * import { Routes, RouterModule } from '@angular/router';
+ *
+ * const routes: Routes = [{
+ *   path: 'lazy',
+ *   loadChildren: () => import('./lazy/lazy.module').then(m => m.LazyModule).
+ * }];
+ *
+ * @NgModule({
+ *   imports: [RouterModule.forRoot(routes)],
+ *   exports: [RouterModule]
+ * })
+ * export class AppRoutingModule { }
+ *
+ * NGC (View Engine) will process it into:
+ *
+ * import { Routes } from '@angular/router';
+ * const ɵ0 = () => import('./lazy/lazy.module').then(m => m.LazyModule);
+ * const routes: Routes = [{
+ *         path: 'lazy',
+ *         loadChildren: ɵ0
+ *     }];
+ * export class AppRoutingModule {
+ * }
+ * export { ɵ0 };
+ *
+ * The importFactory transformation will only see the AST after it is process by NGC.
+ * You can confirm this with the code below:
+ *
+ * const res = ts.createPrinter().printNode(ts.EmitHint.Unspecified, sourceFile, sourceFile);
+ * console.log(`### Original source: \n${sourceFile.text}\n###`);
+ * console.log(`### Current source: \n${currentText}\n###`);
+ *
+ * At this point it doesn't yet matter what the target (ES5/ES2015/etc) is, so the original
+ * constructs, like `class` and arrow functions, still remain.
+ *
+ */
+
+export function importFactory(
+  warningCb: (warning: string) => void,
+): ts.TransformerFactory<ts.SourceFile> {
+  return (context: ts.TransformationContext) => {
+    return (sourceFile: ts.SourceFile) => {
+      const warning = `
+Found 'loadChildren' with a non-string syntax in ${sourceFile.fileName} but could not transform it.
+Make sure it matches the format below:
+
+loadChildren: () => import('IMPORT_STRING').then(m => m.EXPORT_NAME)
+
+Please note that only IMPORT_STRING and EXPORT_NAME can be replaced in this format.`;
```

Comment:
link to the guide again, if ppl get confused we'll be able to help them there

Reply:
Done